activemq-client/src/main/java/org/apache/activemq/ActiveMQProducer.java

```diff
@@ -0,0 +1,547 @@
+                      byte[] array = byte[].class.cast(value);
+                      mapMessage.setBytes(key, array, 0, array.length);
+                  } else {
+                      mapMessage.setObject(key, value);
+                  }
+               }
+            } catch (JMSException e) {
+               throw new MessageFormatRuntimeException(e.getMessage());
+            }
+         }
+         send(destination, mapMessage);
+         return this;
+    }
+
+    @Override
+    public JMSProducer send(Destination destination, byte[] body) {
+        BytesMessage bytesMessage = activemqContext.createBytesMessage();
+
+        try {
+            if(body != null) {
+                bytesMessage.writeBytes(body);
+            }
+            send(destination, bytesMessage);
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+        return this;
+    }
+
+    @Override
+    public JMSProducer send(Destination destination, Serializable body) {
+        ObjectMessage objectMessage = activemqContext.createObjectMessage(body);
+        send(destination, objectMessage);
+        return this;
+    }
+
+    @Override
+    public JMSProducer setDisableMessageID(boolean value) {
+        try {
+            activemqMessageProducer.setDisableMessageID(value);
+            return this;
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean getDisableMessageID() {
+        try {
+            return activemqMessageProducer.getDisableMessageID();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setDisableMessageTimestamp(boolean value) {
+        try {
+            activemqMessageProducer.setDisableMessageTimestamp(value);
+            return this;
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean getDisableMessageTimestamp() {
+        try {
+            return activemqMessageProducer.getDisableMessageTimestamp();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setDeliveryMode(int deliveryMode) {
+        this.deliveryMode = deliveryMode;
+        return this;
+    }
+
+    @Override
+    public int getDeliveryMode() {
+        if(deliveryMode != null) {
+            return deliveryMode;
+        }
+
+        try {
+            return this.activemqMessageProducer.getDeliveryMode();
+        } catch (JMSException e) {
+            throw JMSExceptionSupport.convertToJMSRuntimeException(e);
+        }
+    }
+
+    @Override
+    public JMSProducer setPriority(int priority) {
```

Comment:
Currently, ActiveMQ quietly adjusts out-of-range priority values. I think changing it here could break existing clients:

[o.a.amq.command.Message](https://github.com/apache/activemq/blob/d5441530a9f58163fcca2067f33292697737d9c7/activemq-client/src/main/java/org/apache/activemq/command/Message.java#L424)

Reply:
Existing clients cant be using a JMSProducer implementation that hasnt existed before, so it seems unlikely they will be broken by it validating its values here (not in MessageProducer) as Tim suggested. If they are rewriting their application to a totally new producer API they can fix invalid values at the same time.

Reply:
I don't really see how it would be a breaking change for framework users as there are no users of the ActiveMQ JMS 2.0 client as it doesn't yet exist.  Also there are at least two JMS 2.0 clients that I'm aware of which have JMS 2.0 implementations out in the wild now for **years** which do indeed validate the specification defined values which would mean that the new as yet unreleased JMS 2.0 ActiveMQ 5 bits would be the spanner in the works.

Reply:
What Tim said, echoing what I already said. You cant break something that simply doesnt exist yet. The point of implementing it is the _perfect_ time to ensure it does _what it is meant to_.

Also, whilst I dont recall if it is so in this case, I think a certain conformance suite of interest may actually check _some_ of the things being generally mentioned here are actually validated.

Reply:
I'll add the priority range checking and throw exception if out of bounds. Additionally, I'll add a note to JMS2 documentation to indicate behavior change for JMS v1.x users.

Reply:
This is completed

Reply:
This is completed